sched/signal/sig_dispatch.c

```diff
@@ -424,7 +424,18 @@ int nxsig_tcbdispatch(FAR struct tcb_s *stcb, siginfo_t *info)
 
       if (stcb->task_state == TSTATE_WAIT_SEM)
         {
+#ifndef CONFIG_DISABLE_PTHREAD
```

Comment:
why need check CONFIG_DISABLE_PTHREAD?

Reply:
SIGCONDTIMEDOUT is not defined if CONFIG_DISABLE_PTHREAD=y,
which caused build break.

Reply:
but we should report the correct errno regardless whether pthread is enabled or disabled